doc/harpoon.txt

```diff
@@ -0,0 +1,123 @@
+harpoon.txt*    For Neovim version 0.5                 Last change: 2022 Nov 16
+
+===============================================================================
+CONTENTS                                                     *harpoon-contents*
+
+  Introduction ········································· |harpoon-introduction|
+  Requirements ········································· |harpoon-requirements|
+  Usage ······················································· |harpoon-usage|
+    Commands ··············································· |harpoon-commands|
+    Lua API ················································· |harpoon-lua-api|
+  Configuration ·············································· |harpoon-config|
+  Highlights ············································· |harpoon-highlights|
+  License ··················································· |harpoon-license|
+
+===============================================================================
+INTRODUCTION                                                          *harpoon*
+
+The goal of Harpoon is to get you where you want with the fewest keystrokes.
+
+The Problems:
+    * You're working on a codebase. medium, large, tiny, whatever. You find
+    yourself frequenting a small set of files and you are tired of using a
+    fuzzy finder, :bnext & :bprev are getting too repetitive, alternate file
+    doesn't quite cut it, etc etc.
+    * You want to execute some project specific commands or have any number of
+    persistent terminals that can be easily navigated to.
+
+The Solutions:
+    * The ability to specify, or on the fly, mark and create persisting key
+    strokes to go to the files you want.
+    * Unlimited terminals and navigation.
+
+===============================================================================
+REQUIREMENTS                                             *harpoon-requirements*
+
+Harpoon works only with Neovim and was written with Neovim-0.5, and is
+dependent on plenary.
+
+===============================================================================
+USAGE                                                           *harpoon-usage*
+
+Lua API~
+
+The Lua API comprises several modules. Even though those modules might have
+more public functions than described here, you are only supposed to use the
+functions in this help page. Using one that is not listed here is considered
+unstable.
+
+`harpoon`       Entry point and main interface.
+`harpoon.mark`  Marking files.
+`harpoon.ui`    Ui related commands.
+`harpoon.term`  Neovim term commands.
+`harpoon.tmux`  Tmux term commands.
+
+
+`harpoon.setup(`{opts}`)`                                       *harpoon.setup*
+    Setup Harpoon with options.
+
+    This function will setup the Lua API and commands in a way that respects
```

Comment:
Very verbose. Only necessary thing is that its mandatory to use for all use cases.